File: test/language/templates/python/ExpressionFullEnumTemplateArgumentTest.py
import unittest
import zserio

from testutils import getZserioApi

class ExpressionFullEnumTemplateArgumentTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.api = getZserioApi(__file__, "templates.zs").expression_full_enum_template_argument

    def testReadWrite(self):
        colorInternal = self.api.FullEnumTemplateArgument_Color.fromFields(False, 10)
        colorExternal = (self.api.
                         FullEnumTemplateArgument_templates_expression_full_enum_template_argument_color_Color.
                         fromFields(False, 10))
        fullEnumTemplateArgumentHolder = self.api.FullEnumTemplateArgumentHolder.fromFields(colorInternal,
                                                                                            colorExternal)
        self.assertTrue(fullEnumTemplateArgumentHolder.getEnumTemplateArgumentInternal().hasExpressionField())
        self.assertFalse(fullEnumTemplateArgumentHolder.getEnumTemplateArgumentExternal().hasExpressionField())

        writer = zserio.BitStreamWriter()
        fullEnumTemplateArgumentHolder.write(writer)
        reader = zserio.BitStreamReader(writer.getByteArray())
        readFullEnumTemplateArgumentHolder = self.api.FullEnumTemplateArgumentHolder()
        readFullEnumTemplateArgumentHolder.read(reader)
        self.assertEqual(fullEnumTemplateArgumentHolder, readFullEnumTemplateArgumentHolder)

File: test/language/templates/cpp/ExpressionFullEnumTemplateArgumentTest.cpp
#include "gtest/gtest.h"

#include "templates/expression_full_enum_template_argument/FullEnumTemplateArgumentHolder.h"

namespace templates
{
namespace expression_full_enum_template_argument
{

TEST(ExpressionFullEnumTemplateArgumentTest, readWrite)
{
    const FullEnumTemplateArgument_Color colorInternal(false, 10);
    const FullEnumTemplateArgument_templates_expression_full_enum_template_argument_color_Color colorExternal(
            false, 10);
    FullEnumTemplateArgumentHolder fullEnumTemplateArgumentHolder(colorInternal, colorExternal);
    ASSERT_TRUE(fullEnumTemplateArgumentHolder.getEnumTemplateArgumentInternal().hasExpressionField());
    ASSERT_FALSE(fullEnumTemplateArgumentHolder.getEnumTemplateArgumentExternal().hasExpressionField());

    zserio::BitStreamWriter writer;
    fullEnumTemplateArgumentHolder.write(writer);
    size_t bufferSize = 0;
    const uint8_t* buffer = writer.getWriteBuffer(bufferSize);

    zserio::BitStreamReader reader(buffer, bufferSize);
    const FullEnumTemplateArgumentHolder readFullEnumTemplateArgumentHolder(reader);

    ASSERT_TRUE(fullEnumTemplateArgumentHolder == readFullEnumTemplateArgumentHolder);
}

} // namespace expression_full_enum_template_argument
} // namespace templates
